credit/postblock.py

```diff
@@ -916,36 +925,822 @@ def concat_fix(y_pred, q_pred_correct, q_ind_start, q_ind_end, N_vars):
+        return supported_models[name](**model_conf)
+    else:
+        raise OSError(
+            f"Model name {name} not recognized. Please choose from {supported_models.keys()}"
+        )
+
+class Backscatter_unet(nn.Module):
+    def __init__(self,
+                 in_channels,
+                 levels,
+                 nlat,
+                 nlon,
+                 architecture,
+                 padding):
+        # could also predict with x_prev and y
+        super().__init__()
+        self.nlat = nlat
+        self.nlon = nlon
+        self.in_channels = in_channels
+        self.levels = levels
+        
+        # setup padding functions
+        self.pad = padding
+        if self.pad:
+            logger.info(f"padding size {self.pad} inside unet")
+            self.boundary_padding = TensorPadding(pad_lat=(self.pad, self.pad),
+                                                pad_lon=(self.pad, self.pad))
+
+        self.relu = nn.ReLU()
+
+        if architecture is None:
+            architecture = {
+                            "name": "unet++",
+                            "encoder_name": "resnet34",
+                            "encoder_weights": "imagenet",
+                        }
+        if architecture["name"] == "unet":
+            architecture["decoder_attention_type"] = "scse"
+
+        architecture["in_channels"] = in_channels
+        architecture["classes"] = levels
+
+        self.model = load_premade_encoder_model(architecture)
+    
+    def forward(self, x):
+        x = x.squeeze(2) # squeeze out time dim (see above)
+
+        if self.pad:
+            x = self.boundary_padding.pad(x)
+        
+        x = self.model(x)
+
+        if self.pad:
+            x = self.boundary_padding.unpad(x)
+
+        x = self.relu(x)
+
+
+        x = x.unsqueeze(2)
+        return x
+
+class Backscatter_fixed_col(nn.Module):
+    def __init__(self,
+                 levels,):
+        super().__init__()
+        self.backscatter_array = Parameter(torch.full((1,levels,1,1,1), 2.5))
+
+    def forward(self, x):
+        # self.backscatter_array.data = self.backscatter_array.data.clamp(0., 10.)
+        logger.debug(torch.flatten(self.backscatter_array))
+        return self.backscatter_array  # this will be inside sqrt
+    
+
+class Backscatter_prescribed(nn.Module):
+    def __init__(self,
+                 nlat,
+                 nlon,
+                 levels,
+                 std_path,
+                 sigma_max):
+        super().__init__()
+        self.nlat = nlat
+        self.nlon = nlon
+
+        std = xr.open_dataset(std_path)
+        std_wind_avg = (std.U.values + std.V.values) / 2.
+        self.backscatter_array = Parameter(torch.tensor(1e-2 * (std_wind_avg * sigma_max) ** 2).reshape(1,levels,1,1,1), requires_grad=True)
+        # formula to convert to 1% of sigma * std_wind_avg
+
+    def forward(self, x):
+        return self.backscatter_array
+    
 
 class SKEBS(nn.Module):
```

Comment:
Given how long the postblock.py module is getting, I would highly recommend moving skebs to a separate module, or making postblock.py a directory like models and having separate files for each type of postblock.